src/components/Chat/ChatPanel/ChatForm/ChatForm.tsx

```diff
@@ -15,140 +15,134 @@ import styles from './ChatForm.module.css';
-                                <BsPlusCircleFill className={cx({ disabled: isLoading })} />
-                            )}
-                        </button>
-                        <button type="submit" className={cx('submit-btn', { active: value })} disabled={!value}>
-                            <PiPaperPlaneTiltFill />
-                        </button>
-                    </div>
-                </form>
-            </div>
-            <input
-                type="file"
-                accept="/image/*"
-                ref={fileInputRef}
-                onChange={handleChange}
-                style={{ display: 'none' }}
-            />
-        </div>
-    );
+    return message;
+  };
+
+  //>>>>>>>>>>>>>>>>>>>>>>>>>>파일 업로드
+  const handleOpenFile = () => {
+    assert(fileInputRef.current !== null, '파일 참조가 없습니다.');
+    fileInputRef.current.click();
+  };
+
+  const handleChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    const { files } = event.target;
+    const file = files && files[0];
+    assert(file !== null, '업로드 된 파일이 없습니다.');
+    setIsLoading(true);
+    try {
+      const downloadUrl = await uploadImage(chatRoomId, file);
+      addMessage({ chatRoomId, message: createMessage({ file: downloadUrl }) });
+      setIsLoading(false);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  //<<<<<<<<<<<<<<<<<<<<<<<<<<파일 업로드
+
+  //입력된 글자에 따라 teatarea height 조정
+  const resizeTextareaHeight = useCallback(
+    (value: string) => {
+      const lineHeight = 17;
+      const totalTextHeight = value.split('\n').length * lineHeight;
+      const newHeight = totalTextHeight < lineHeight ? textareaHeight : totalTextHeight;
+      setTextareaHeight(newHeight);
+    },
+    [textareaHeight],
+  );
+
+  //텍스트 메시지 전송
+  const handleAddMessage: SubmitHandler<FormData> = (data) => {
+    const content = data.content;
+
+    if (!content || content.trim() === '') return;
+
+    try {
+      addMessage({ chatRoomId, message: createMessage({ content }) });
+    } catch (err) {
+      console.log(err);
+    }
+
+    setTextareaHeight(17);
+    reset();
+  };
+
+  //Enter Submit
+  const handleEnterSubmit: KeyboardEventHandler<HTMLTextAreaElement> = (e) => {
+    if (e.key !== 'Enter') return;
+    if (e.shiftKey) return;
+    if (!e.nativeEvent.isComposing) {
+      e.preventDefault();
+      handleAddMessage({
+        content: value,
+      });
+    }
+  };
+
+  return (
+    <div className={cx('container')}>
+      <div className={cx('inner-form-container')}>
+        <form onSubmit={handleSubmit(handleAddMessage)}>
+          <textarea
+            className={cx('text-input')}
+            style={{ height: textareaHeight + 'px' }}
+            {...register('content', {
+              onChange: () => resizeTextareaHeight(getValues('content')),
+            })}
+            rows={1}
+            onKeyDown={handleEnterSubmit}
+            placeholder="내용을 입력해주세요"
+            autoFocus
+          ></textarea>
```

Comment:
```suggestion
          <textarea
            className={cx('text-input')}
            style={{ height: textareaHeight + 'px' }}
            {...register('content', {
              onChange: () => resizeTextareaHeight(getValues('content')),
            })}
            rows={1}
            onKeyDown={handleEnterSubmit}
            placeholder="내용을 입력해주세요"
            autoFocus
          />
```

사소하지만 요런것도 챙기는게 좋아용
기존과 같이 작성하면 children을 쓰려고 했는데 개발자가 실수로 깜빡한 것인지
children이 원래 없었던건지 알수가 없어서
개발자의 의도가 분명해지지 않는다는 단점이 있어요!

Reply:
다른 사람이 작성한 코드를 읽을때도 생각해볼 수 있는 부분이네요!👍